resolwe_bio/processes/import_data/expressions.py

```diff
@@ -0,0 +1,441 @@
+        )
+        feature_type = StringField(
+            label="Feature type",
+            allow_custom_choice=True,
+            default="gene",
+            choices=[
+                ("gene", "gene"),
+                ("transcript", "transcript"),
+                ("exon", "exon"),
+            ],
+        )
+
+    class Output:
+        """Output fields to process UploadExpression."""
+
+        exp = FileField(label="Normalized expression")
+        rc = FileField(
+            label="Read counts",
+            required=False,
+            description="Reads mapped to genomic features.",
+        )
+        exp_json = JsonField(label="Expression (json)")
+        exp_type = StringField(label="Expression type")
+        exp_set = FileField(label="Expressions")
+        exp_set_json = JsonField(label="Expressions (json)")
+        source = StringField(label="Gene ID source")
+        species = StringField(label="Species")
+        build = StringField(label="Build")
+        feature_type = StringField(label="Feature type")
+
+    def run(self, inputs, outputs):
+        """Run analysis."""
+
+        supported_extensions = (".txt", ".tab", ".tsv", "")
+
+        if not inputs.exp and not inputs.rc:
+            self.error("Please provide raw or/and normalized expression files.")
+
+        elif inputs.exp and not inputs.exp_type:
+            self.error(
+                "Please provide normalization type together with normalized expressions."
+            )
+
+        elif not inputs.exp and inputs.exp_type and inputs.rc:
+            self.error("Please provide raw or/and normalized expression files.")
+
+        elif inputs.rc and not inputs.exp and not inputs.exp_type:
+            rc = inputs.rc.import_file(imported_format="compressed")
+            exp = inputs.rc.import_file(imported_format="compressed")
+            exp_type = "RAW_COUNT"
+            stem = Path(rc).stem
+
+        elif inputs.exp and inputs.exp_type and not inputs.rc:
+            exp = inputs.exp.import_file(imported_format="compressed")
+            stem = Path(exp).stem
+            exp_type = inputs.exp_type
+
+        else:
+            rc = inputs.rc.import_file(imported_format="compressed")
+            exp = inputs.exp.import_file(imported_format="compressed")
+            stem = Path(rc).stem
+            stem_exp = Path(exp).stem
+            if not stem_exp.endswith(supported_extensions):
+                self.error(
+                    f"The imported file has unsupported file name extension. "
+                    f"The supported extensions are {supported_extensions}."
+                )
+            exp_type = inputs.exp_type
+
+        if not stem.endswith(supported_extensions):
+            self.error(
+                f"The imported file has unsupported file name extension. "
+                f"The supported extensions are {supported_extensions}."
+            )
+        name = stem[:-4]
```

Comment:
If in either the `rc` or `exp` field, a file without extension (e.g. `my_expressions`) is uploaded with `rc = inputs.rc.import_file(imported_format="compressed")`, then the `stem = Path(rc).stem` would return `my_expressions`, correct? Thus, `name = stem[:-4]` would unintentionally truncate the file name.